qutebrowser/browser/commands.py

```diff
@@ -1443,8 +1443,23 @@ def download(self, url=None, dest_old=None, *, mhtml_=False, dest=None):
                 download_manager.get_mhtml(tab, target)
         else:
             qnam = tab.networkaccessmanager()
-            download_manager.get(self._current_url(), user_agent=user_agent,
-                                 qnam=qnam, target=target)
+
+            # Downloads of URLs with file extensions in the whitelist will use
+            # the page title as the filename.
+            ext_whitelist = [".html", ".htm", ".php", ""]
+            _, ext = os.path.splitext(self._current_url().path())
+            if ext.lower() in ext_whitelist and tab.title():
+                suggested_fn = utils.sanitize_filename(tab.title()) + ext
```

Comment:
When QtWebEngine auto-generates a title, this ends up as `no title.html.html`. I'd suggest checking for `if not tab.title().endswith(ext):` before adding it.